utils/jwtHelper.js

```diff
@@ -0,0 +1,77 @@
+/* eslint-disable camelcase*/
+const KJUR = require("jsrsasign");
+const fs = require("fs");
+const { sanitiseMetadata } = require("../utils/sanitiseMetadata");
+const JSONWebKey = require("json-web-key");
+const { JWK, JWE } = require("node-jose");
+const { find, keys, assign, flow, map } = require("lodash/fp");
+const yaml = require("js-yaml");
+const SIGNING_ALGORITHM = "RS256";
+const keysFile = process.env.KEYS_FILE || "./keys.yml";
+const signingPassword = process.env.SIGNING_PASSWORD || "digitaleq";
+
+const getKeyByUse = (json, useCase) =>
+  flow(
+    keys,
+    map(kid => assign(json[kid], { kid })),
+    find(keyObject => keyObject.use === useCase)
+  )(json);
+
+const keysYaml = yaml.safeLoad(fs.readFileSync(keysFile, "utf8"));
+const keysJson = JSON.parse(JSON.stringify(keysYaml));
+
+module.exports.generateToken = function(questionnaireId, metadataValues) {
+  const signingKeyObject = getKeyByUse(keysJson.keys, "signing");
+  const encryptionKeyObject = getKeyByUse(keysJson.keys, "encryption");
+
+  // Header
+  const jwtHeader = JSON.stringify({
+    alg: SIGNING_ALGORITHM,
+    typ: "JWT",
+    kid: signingKeyObject.kid
+  });
+
+  const tokenIssuedTime = KJUR.jws.IntDate.get("now");
+
+  const payload = sanitiseMetadata(
```

Comment:
Could we pass this all down to this function instead of it knowing anything about metadata sanitising. It feels like the rest of the function is token specific and this payload generation isn't. So we would call sanitiseMetadata in the launch middleware

Reply:
I did have it set-up like that originally however it meant we needed to import the KJUR module both here for the crypto stuff and then also in the launch middleware for the expiry time. I moved it here to try to isolate the crypto imports to one module but understand it makes this function a little less specific. Not sure which approach would be better here?